src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.OperationVisitor.cs

```diff
@@ -35,25 +62,57 @@ private sealed class OperationVisitor : GlobalFlowStateDataFlowOperationVisitor
             {
                 var value = base.VisitInvocation_NonLambdaOrDelegateOrLocalFunction(method, visitedInstance, visitedArguments, invokedAsDelegate, originalOperation, defaultValue);
 
-                if (_platformCheckMethods.Contains(method.OriginalDefinition))
+                if (method.ReturnType.SpecialType == SpecialType.System_Boolean)
                 {
-                    using var infosBuilder = ArrayBuilder<PlatformMethodValue>.GetInstance();
-                    if (PlatformMethodValue.TryDecode(method, visitedArguments, DataFlowAnalysisContext.ValueContentAnalysisResult, _osPlatformType, infosBuilder))
+                    if (_platformCheckMethods.Contains(method.OriginalDefinition))
                     {
-                        for (var i = 0; i < infosBuilder.Count; i++)
+                        using var infosBuilder = ArrayBuilder<PlatformMethodValue>.GetInstance();
+                        if (PlatformMethodValue.TryDecode(method, visitedArguments, DataFlowAnalysisContext.ValueContentAnalysisResult, _osPlatformType, infosBuilder))
                         {
-                            var newValue = GlobalFlowStateAnalysisValueSet.Create(infosBuilder[i]);
-                            value = i == 0 ? newValue : GlobalFlowStateAnalysis.GlobalFlowStateAnalysisValueSetDomain.Instance.Merge(value, newValue);
+                            for (var i = 0; i < infosBuilder.Count; i++)
+                            {
+                                var newValue = GlobalFlowStateAnalysisValueSet.Create(infosBuilder[i]);
+                                value = i == 0 ? newValue : GlobalFlowStateAnalysis.GlobalFlowStateAnalysisValueSetDomain.Instance.Merge(value, newValue);
```

Comment:
The inconsistency between the code here and the newly added one above which uses `WithAdditionalAnalysisValues` for negated values makes me cringe a bit more. IMO, the analysis should be identical when decoding method attributes at callsites, regardless of whether it is a well-known platform check method OR non-well known method with the new guard attributes.

Reply:
It could be identical if annotation made same as guard methods, i, e, only check if a platform is supported. But in this case we have guards for Unsupported which i don't think we can efficiently express same as existing Guard methods, well we could somehow express them same us guard methods but that would be much more complicated